### PR TITLE
feat(packages): add csv-price-parser package

### LINKED ISSUE #27: Move csv-parser-price to this repository
Move the [`csv-parser-price`](https://github.com/commercetools/csv-parser-price/) module to this repository.

Doing this will also require changes to the code of the module and some additional steps:
- [x] Use the new SDK packages as dependencies
- [x] Add a `types` endpoint to the SDK, [PR](https://github.com/commercetools/nodejs/pull/24)
- [x] Get tests passing manually passing credentials
- [x] Add a package to get credentials from the FS
- [x] Check if everything is working as expected manually using the CLI
- [x] Get rid of underscore dependencies
- [x] Convert tape to jest tests
*Got this done by using `jest-codemods`, replacing `it` for `test` and then manually grouping all same-function tests in describe blocks and rewriting `.then().catch()` chains.*
- [x] Remove all root files that already exist in the monorepo root
- [x] Slim down `package.json` fields and development dependencies
- [x] Migrate CLI tests that talk to the API to the integration tests directory
- [x] Refactor arguments to be named
*Accepting one object instead of three unnamed objects.*

### PR DESCRIPTION
**Pull request by Siilwyn**
affects: @commercetools/csv-parser-price

Resolves #27
BREAKING CHANGE: migration to new SDK

### CODE REVIEW

File: packages/csv-parser-price/readme.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                                               [default: "info"]
+```
+When running with `--outputFile` set, logging information is shown directly otherwise it goes to `csvparserprice.log` with the exception of fatal errors.
+
+### JS
+```js
+const fs = require('fs');
+const CsvParserPrice = require('@commercetools/csv-parser-price');
+
+const csvParserPrice = new CsvParserPrice(
```
Comment (adnasa) [inline packages/csv-parser-price/readme.md]: Inconsistent indentation in this example

File: packages/csv-parser-price/src/map-custom-fields.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            }
+          }
+      })
+    })
+    return result
+  }
+
+  function mapBoolean (value) {
+    const result = {}
+    if (value === undefined || (typeof value === 'string' && value === ''))
```
Comment (adnasa) [inline packages/csv-parser-price/src/map-custom-fields.js]: I'm pretty sure that this if-statement would need to be re-used elsewhere. can't this be extracted out into a separate util?

```js
if (isEmpty(value)) {

}
```

the naming needs work in my example, but I think you get what I mean 😉

File: packages/csv-parser-price/src/cli.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  else
+    logError(errors)
+
+  process.exit(1)
+}
+
+// TODO: where is the accessToken gone?
+const csvParserPrice = new CsvParserPrice(
+  {
+    // TODO: make this an option
```
Comment (wizzy25) [inline packages/csv-parser-price/src/cli.js]: Should you be merging ```TODO``` comments? 🤓

File: packages/csv-parser-price/test/cli.spec.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import { exec } from 'child_process'
+import fs from 'fs'
+import tmp from 'tmp'
+import CONSTANTS from '../src/constants'
+import { version } from '../package.json'
+
+
+const binPath = './bin/csvparserprice.js'
+
+// TODO: replace with package call
```
Comment (wizzy25) [inline packages/csv-parser-price/test/cli.spec.js]: More ```TODO```'s.....

File: packages/csv-parser-price/test/main.spec.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+import customTypeSample from './helpers/custom-type-sample.json'
+
+const logger = {
+  error: () => {},
+  warn: () => {},
+  info: () => {},
+  verbose: () => {},
+}
+
+// TODO: replace with package call (sdk-middleware-auth-password)
```
Comment (wizzy25) [inline packages/csv-parser-price/test/main.spec.js]: 🙂

**Comment by Siilwyn** [comment]
@wizzy25 please remove your review. This is still a work in progress branch, hence the label and no requested reviewers, I use those comments to keep track of my progress. :|

**Comment by wizzy25** [comment]
This build would fail until #87 gets merged... 🤔

**Comment by Siilwyn** [comment]
Since this changes the ENV variable: do not forget to remove 'CREDENTIALS_NODE_SDK_INTEGRATION_TESTS' from Travis when this is merged!!! @wizzy25 @hisabimbola

File: packages/csv-parser-price/readme.md (hunk 1 above)
Comment (wizzy25) [inline packages/csv-parser-price/readme.md]: @adnasa 8a8ca2d

**Comment by wizzy25** [comment]
LGTM 👍🏾

**Comment by Siilwyn** [comment]
Don't forget:
- editing the readme: moving documentation
- deprecating the csv-parser-price repo
- this is a major bump
- https://github.com/sphereio/sphere-product-import/pull/97
- https://github.com/sphereio/sphere-node-cli/pull/56

~ Over & out. :roller_coaster:

File: integration-tests/sdk/channels.it.js [hunk 6]
```diff
@@ -14,50 +15,47 @@ import {
...
-  middlewares: [
-    authMiddleware,
-    queueMiddleware,
-    userAgentMiddleware,
-    httpMiddleware,
-  ],
-})
 
 describe('Channels', () => {
+  const projectKey = 'node-sdk-integration-tests'
```
Comment (hisabimbola) [inline integration-tests/sdk/channels.it.js]: Do we have to hard-code this??

File: packages/csv-parser-price/my-prices.csv [hunk 7]
```diff
@@ -0,0 +1,4 @@
+variant-sku,value.currencyCode,value.centAmount,customType,customField.foo
```
Comment (hisabimbola) [inline packages/csv-parser-price/my-prices.csv]: I think we should group this in a folder called 'samples' or something, not just putting it in the root

Comment (hisabimbola) [inline packages/csv-parser-price/my-prices.csv]: I will actually prefer we use the docs for this

File: packages/csv-parser-price/output.json [hunk 8]
```diff
@@ -0,0 +1 @@
+[{"variant-sku":"do-price","prices":[{"variant-sku":"do-price","value":{"centAmount":4000},"custom":{"type":{"id":"664f5681-e0d1-442f-a241-acfb6f1d7bce"},"fields":{"foo":true}}}]},{"variant-sku":"my-price","prices":[{"variant-sku":"my-price","value":{"centAmount":3000},"custom":{"type":{"id":"664f5681-e0d1-442f-a241-acfb6f1d7bce"},"fields":{"foo":true}}},{"variant-sku":"my-price","value":{"centAmount":4000},"custom":{"type":{"id":"664f5681-e0d1-442f-a241-acfb6f1d7bce"},"fields":{"foo":false}}}]}]
```
Comment (hisabimbola) [inline packages/csv-parser-price/output.json]: Same as above...

File: packages/csv-parser-price/readme.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+[![commercetools logo][commercetools-icon]][commercetools]
```
Comment (hisabimbola) [inline packages/csv-parser-price/readme.md]: We can do like other packages and link the page in the official docs for this insteand...

File: packages/csv-parser-price/src/constants.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  },
+
+  standardOption: {
+    batchSize: 100,
+    delimiter: ',',
+  },
+}
+
+// Go through object because `freeze` works shallow
+Object.keys(CONSTANTS).forEach((key) => {
```
Comment (hisabimbola) [inline packages/csv-parser-price/src/constants.js]: My inner spirit tells me there is a module that does it, so should be refactored to a module if non exist 

👻

**Comment by hisabimbola** [comment]
I think this is ready to be merged!!! 🚀

**Comment by hisabimbola** [comment]
I guess there is no need for a major bump again, infact, no need for a release

We just migrated this module from a location to another

I am going to be merge this PR using `chore`

**Pull request merged**